pkg/controller/operators/openshift/helpers.go

```diff
@@ -105,58 +131,59 @@ func incompatibleOperators(ctx context.Context, cli client.Client) (skews, error
 	}
 
 	if next == nil {
-		return nil, nil
+		// Note: This shouldn't happen
+		return nil, fmt.Errorf("Failed to determine next OpenShift Y-stream release")
 	}
 	next.Minor++
 
 	csvList := &operatorsv1alpha1.ClusterServiceVersionList{}
 	if err := cli.List(ctx, csvList); err != nil {
-		return nil, err
+		return nil, &transientError{fmt.Errorf("Failed to list ClusterServiceVersions: %w", err)}
 	}
 
-	var (
-		s    skews
-		errs []error
-	)
+	var incompatible skews
 	for _, csv := range csvList.Items {
 		if csv.IsCopied() {
 			continue
 		}
 
+		s := skew{
+			name:      csv.GetName(),
+			namespace: csv.GetNamespace(),
+		}
 		max, err := maxOpenShiftVersion(&csv)
 		if err != nil {
-			errs = append(errs, err)
+			s.err = err
+			incompatible = append(incompatible, s)
 			continue
 		}
+
 		if max == nil || max.GTE(*next) {
 			continue
 		}
+		s.maxOpenShiftVersion = max.String()
 
-		s = append(s, skew{
-			name:                csv.GetName(),
-			namespace:           csv.GetNamespace(),
-			maxOpenShiftVersion: max.String(),
-		})
+		incompatible = append(incompatible, s)
 	}
 
-	return s, utilerrors.NewAggregate(errs)
+	return incompatible, nil
 }
 
 func desiredRelease(ctx context.Context, cli client.Client) (*semver.Version, error) {
 	cv := configv1.ClusterVersion{}
 	if err := cli.Get(ctx, client.ObjectKey{Name: "version"}, &cv); err != nil { // "version" is the name of OpenShift's ClusterVersion singleton
-		return nil, err
+		return nil, &transientError{fmt.Errorf("Failed to get ClusterVersion: %w", err)}
```

Comment:
One thing that just crossed my mind: Is this section of code disabled in non-OpenShift clusters? I assume `maxOpenshiftVersion` is just ignored unless we're in Openshift?

Reply:
It looks like the only call site for this function is in `incompatibleOperators`, and that function is only used in the context of the ClusterOperator controller that handles the maxOpenshiftVersion logic.

I guess we gate the creation of this controller on the [`--writeStatusName` CLI flag](https://github.com/operator-framework/operator-lifecycle-manager/blob/b044fdcccb00bc7859956ca3f3500d85b3af862f/cmd/olm/main.go#L226) that we set downstream, which seems like a relatively loose check (couldn't we discovery this API dynamically instead?), so tl;dr should be ignored on non-OCP clusters.

Reply:
@joelanford, @timflannagan hit the nail on the head: the entire controller is disabled unless `--writeStatusName` is given when running `olm-operator`.